test/examples/test_examples.py

```diff
@@ -25,15 +61,100 @@ def test_basic_configuration():
+    #('--keep-last 10 --max-samples-per-instance 1', ''), # this one only displays warning from now on
+    ('--liveliness-kind AUTOMATIC', '--liveliness-kind MANUAL_BY_PARTICIPANT'),   # Liveliness QoS
+    ('--liveliness-kind AUTOMATIC', '--liveliness-kind MANUAL_BY_TOPIC'),
+    ('--liveliness-kind MANUAL_BY_PARTICIPANT', '--liveliness-kind MANUAL_BY_TOPIC'),
+    ('--ownership', ''),                                                           # Ownership QoS
+    ('', '--ownership'),
+    ('--partition configuration_example_partition', '--partition other_partition'), # Partition QoS
+    ('', '--reliable'),                                                          # Reliability QoS
+]
+
+@pytest.mark.parametrize("pub_args, sub_args", timeout_test_cases)
+def test_configuration_timeout(pub_args, sub_args):
+    """."""
+    ret = False
+    out = ''
+
+    command_prerequisites = 'PUB_ARGS="' + pub_args + '" SUB_ARGS="' + sub_args + '" '
+    try:
+        out = subprocess.check_output(command_prerequisites + '@DOCKER_EXECUTABLE@ compose -f configuration.compose.yml up',
+            stderr=subprocess.STDOUT,
+            shell=True,
+            timeout=10
+        )
+    except subprocess.CalledProcessError as e:
+        print (e.output)
+    except subprocess.TimeoutExpired:
+        ret = True
+        subprocess.check_output('@DOCKER_EXECUTABLE@ compose -f configuration.compose.yml down',
+            stderr=subprocess.STDOUT,
+            shell=True,
+            timeout=15
+        )
+
+    assert(ret)
+
+expected_output_test_cases = [
+    ('--deadline 80', '--deadline 80', 'Deadline missed!', '10'),               # 10 = n samples sent
+    ('--deadline 80', '--deadline 80', 'Requested deadline missed!', '15'),     # 20 = n samples received (10 x 2) (flaky)
+]
+
+@pytest.mark.parametrize("pub_args, sub_args, expected_message, n_messages", expected_output_test_cases)
+def test_configuration_expected_output(pub_args, sub_args, expected_message, n_messages):
+    """."""
+    ret = False
+    out = ''
+    pub_requirements = '--reliable --transient-local --keep-last 10'
+    sub_requirements = '--reliable --transient-local --keep-last 10'
+
+    command_prerequisites = 'PUB_ARGS="' + pub_requirements + ' ' + pub_args + '" SUB_ARGS="' + sub_requirements + ' ' + sub_args + '" '
+    try:
+        out = subprocess.check_output(command_prerequisites + '@DOCKER_EXECUTABLE@ compose -f configuration.compose.yml up',
+            stderr=subprocess.STDOUT,
+            shell=True,
+            timeout=30
+        )
+        render_out = out.decode().split('\n')
+
+        count = 0
+        for line in render_out:
+            if expected_message in line:
+                count += 1
+                continue
+
+        if count >= int(n_messages):
             ret = True
         else:
-            raise subprocess.CalledProcessError(1, '')
+            print ('ERROR: expected: 10 "' + expected_message + '" messages, but received ' + str(count))
```

Comment:
This should be updated

Reply:
(Also, remove `-d 113`  from pub/sub_requirements)